craigslist/__init__.py

```diff
@@ -237,25 +247,53 @@ def customize_result(self, result, html_row):
         """ Add custom/delete/alter fields to result. """
         pass  # Override in subclass to add category-specific fields.
 
-    def geotag_result(self, result):
+    def geotag_result(self, result, soup):
         """ Adds (lat, lng) to result. """
 
         self.logger.debug('Geotagging result ...')
 
-        if result['has_map']:
-            response = requests_get(result['url'], logger=self.logger)
-            self.logger.info('GET %s', response.url)
-            self.logger.info('Response code: %s', response.status_code)
-
-            if response.ok:
-                soup = BeautifulSoup(response.content, 'html.parser')
-                map = soup.find('div', {'id': 'map'})
-                if map:
-                    result['geotag'] = (float(map.attrs['data-latitude']),
-                                        float(map.attrs['data-longitude']))
+        map = soup.find('div', {'id': 'map'})
+        if map:
+            result['geotag'] = (float(map.attrs['data-latitude']),
+                                float(map.attrs['data-longitude']))
 
         return result
 
+    def include_details(self, result, soup):
+        """ Adds description, images to result """
+
+        self.logger.debug('Adding details to result...')
+
+        body = soup.find('section', {'id': 'postingbody'})
+        result['body'] = body.text.strip()
+
+        img_matcher = re.compile(
```

Comment:
Any reason why this particularly needs regular expressions? I think this can be simplified by the assumption that all image URLs will have the predefined format. I'd do something along the lines of:

```python
image_urls = set()
for img in soup.find_all('img'):
    url = img['src'].rsplit('_')[0] + '_600x450.jpg'
    # or even: url = img['src'].replace('50x50c', '_600x450')
    images_urls.add(url)
result['images'] = list(images)
```